rust/src/pgsql/logger.rs

```diff
@@ -0,0 +1,432 @@
+/* Copyright (C) 2021 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+// Author: Juliana Fajardini <jufajardini@gmail.com>
+
+//! PostgreSQL parser json logger
+
+use std;
+use crate::jsonbuilder::{JsonBuilder, JsonError};
+use crate::pgsql::pgsql::*;
+use crate::pgsql::parser::*;
+
+fn log_pgsql(tx: &PgsqlTransaction, js: &mut JsonBuilder) -> Result<(), JsonError> {
+    js.set_uint("transaction_id", tx.tx_id.into())?;
+    if !tx.requests.is_empty() {
+        if tx.requests.len() > 1 {
+            js.open_array("requests")?;
+            for request in &tx.requests {
+                let jbr = log_request(&request)?;
+                js.append_object(&jbr)?;
```

Comment:
Should be able to avoid new object allocation, and just build onto the `js`. If not, please comment why.